src/mgos_pcf857x_internal.h

```diff
@@ -46,6 +46,16 @@ struct mgos_pcf857x {
   struct mgos_pcf857x_cb cb[16];
 };
 
+struct mgos_pcf857x_gpio_blink_state {
+  int pin;
+  struct mgos_pcf857x *dev;
+  struct {
```

Comment:
why do you create anested blink struct here, and not just directly fold here?

Reply:
You're right. It has been refactored.